bip155: mark torv2 as deprecated

brunoerg opened the pull request:
TorV2 is deprecated, implementations are removing any support for it and we could specify this deprecation here. Especially because implementations are no longer serializing/deserializing and relaying Tor v2 addresses, they're ignoring incoming Tor v2 addresses, etc.

jonatack commented:
Pinging BIP author @laanwj for feedback or approval.

Not sure if Appendix A about Tor v2 address encoding should be removed or kept for historical context.

jonatack commented:
Perhaps a Changelog would be useful here.

jonatack commented:
Maybe `Appendix A: Tor v2 address encoding (deprecated and no longer operational)`

murchandamus commented on bip-0155.mediawiki:
```diff
@@ -101,7 +101,7 @@ The list of reserved network IDs is as follows:
 | <code>TORV2</code>
 | 10
-| Tor v2 hidden service address
+| Tor v2 hidden service address (**Tor v2 is no longer operational**; clients **MUST NOT** gossip or relay Tor v2 addresses and **MUST** ignore them on receive).
```
The MediaWiki format uses single quotes to format text as follows: ```''italic''```, ```'''bold'''```, ```'''''bold & italic'''''```.

brunoerg replied:
Good to know, thanks.

murchandamus reviewed (commented):
Thanks for the review, @laanwj.
@brunoerg: I think adding a Changelog section and moving the majority of the text into a footnote sounds good.

brunoerg commented:
Just moved the text into a footnote (I just noted that some BIPs calls the section that refers to `<references/>` as `References` and some others as `Footnotes` - so I think I put it on the right place) and added a `Changelog`.

CI [failure](https://github.com/bitcoin/bips/actions/runs/18171090099/job/51725721646?pr=1975) seems unrelated:
```
./typos
jq: jq-1.7
$ ./typos .
Warning: "Falke" should be "False".
error: `Falke` should be `False`
    ╭▸ ./bip-0099.mediawiki:355:62
    │
355 │ Btcdrak, Gavin Andresen, Gregory Sanders, Luke Dashjr, Marco Falke.
```

brunoerg commented on bip-0155.mediawiki:
```diff
@@ -101,7 +101,7 @@ The list of reserved network IDs is as follows:
 | <code>TORV2</code>
 | 10
-| Tor v2 hidden service address
+| Tor v2 hidden service address (no longer used<ref>Tor v2 is no longer operational; clients MUST NOT gossip or relay Tor v2 addresses and MUST ignore them on receive</ref>)
```
I agree it's odd to leave the Appendix A as is. We could point out there to the same `<ref>` we introduced before.

brunoerg replied:
Done.

murchandamus commented on bip-0155.mediawiki:
```diff
@@ -152,7 +152,7 @@ The reference implementation is available at (to be done)
 - Gregory Maxwell: various suggestions regarding extensibility
 
-==Appendix A: Tor v2 address encoding==
+==Appendix A: Tor v2 address encoding (no longer used <ref name="torv2" />)==
```
Nit: the space here should be dropped.

```suggestion
==Appendix A: Tor v2 address encoding (no longer used<ref name="torv2" />)==
```

murchandamus reviewed (commented):
Looks good to me.

murchandamus commented on bip-0155.mediawiki:
```diff
@@ -193,6 +193,13 @@ Cjdns addresses are simply IPv6 addresses in the <code>fc00::/8</code> range<ref
 
+== Changelog ==
+
+* 0.2.0 (2025-10-01):
+** Add note that Tor v2 is no longer operational.
+* 0.1.0 (2019-02-27):
+** Initial version
```
Actually, per BIP3 which specifies the Changelog section, the initial version should be 1.0.0. And this may well be considered a breaking change of the specification?

jonatack replied:
Yes, 1.0.0 and 2.0.0 SGTM. @brunoerg mind updating?
